onecodex/cli.py

```diff
@@ -186,12 +186,26 @@ def samples(ctx, samples):
 @click.option('--prompt/--no-prompt', is_flag=True, help=OPTION_HELP['prompt'], default=True)
 @click.option('--validate/--do-not-validate', is_flag=True, help=OPTION_HELP['validate'],
               default=True)
+@click.option('--tags', type=str)
+@click.option("--metadata", '-md', multiple=True)
 @click.pass_context
 @telemetry
 def upload(ctx, files, max_threads, clean, no_interleave, prompt, validate,
-           forward, reverse):
+           forward, reverse, tags, metadata):
     """Upload a FASTA or FASTQ (optionally gzip'd) to One Codex"""
 
+    tag_array = []
+    if tags:
+        tag_array = [tag.strip() for tag in tags.split(',')]
+
+    metadata_opts = {}
+    if metadata:
+        for metadata_kv in metadata:
+            split_metadata = metadata_kv.split('=')
+            if len(split_metadata) > 1:
+                metadata_value = '='.join(split_metadata[1:])
+                metadata_opts[split_metadata[0].lower()] = metadata_value
```

Comment:
We should consider forcing metadata field names into snakecase (we do this within the One Codex web app).